system/system.go

```diff
@@ -82,28 +82,90 @@ func (c *Cmd) StdoutPipe() (io.ReadCloser, error) {
 	return c.Cmd.StdoutPipe()
 }
 
+func Command(name string, arg ...string) *Cmd {
+	var cmd Cmd
+	cmd.Cmd = exec.Command(name, arg...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return &cmd
+}
+
 type cmdLogger struct {
-	cmdName string
-	stream  string
+	commandName string
+	stream      string
+	*bytes.Buffer
+}
+
+func NewCmdLoggerStdout(cmd string) *cmdLogger {
+	return &cmdLogger{
+		commandName: cmd,
+		stream:      "stdout",
+		Buffer:      bytes.NewBuffer(nil),
+	}
+}
+
+func NewCmdLoggerStderr(cmd string) *cmdLogger {
+	return &cmdLogger{
+		commandName: cmd,
+		stream:      "stderr",
+		Buffer:      bytes.NewBuffer(nil),
+	}
+}
+
+func (c *cmdLogger) Write(b []byte) (int, error) {
+	n, err := c.Buffer.Write(b)
+	c.writeLines()
+	return n, err
 }
 
-func (c *cmdLogger) Write(buf []byte) (int, error) {
-	lines := strings.Split(string(buf), "\n")
-	for _, line := range lines {
-		if len(line) > 0 {
-			log.Infof("Output (%s) from command %q: %s", c.stream, c.cmdName, line)
+func (c *cmdLogger) writeLines() {
+	if bytes.Contains(c.Buffer.Bytes(), []byte{'\n'}) {
+		line, lerr := c.Buffer.ReadString('\n')
+		if lerr != nil {
+			panic("This should never happen unless we have a race condition!")
 		}
+		log.Infof(
+			"Output (%s) from command %q: %s",
+			c.stream,
+			c.commandName,
+			line[:len(line)-1], // Do not write the newline delimeter
+		)
 	}
+}
 
-	return len(buf), nil
+func (c *cmdLogger) Flush() {
+	c.writeLines()
+	// Empty the remaining bytes
```

Comment:
A hack could be to add a newline, and then call writelines, but it felt a tad hacky

Reply:
Agree!